### PR TITLE
shift & unshift support

### PR DESCRIPTION
**Pull request by michalbe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,6 +78,63 @@ output; // {foo: {bar: {baz: ['a', 'b']}}}
 
 You can find more examples in the test files.
 
+### `unshift`
+
+Immutably unshiftes (is that even a word?) the given values to the respective part of the input data structure.
```
Comment (ennovum) [inline README.md:83]: No, it's not a word. `inserts ... to the beginning of the array`?

File: README.md [hunk 2]
```diff
@@ -78,6 +78,63 @@ output; // {foo: {bar: {baz: ['a', 'b']}}}
...
+
+```
+output = immutably.array.unshift(input, path, ...values);
+```
+
+**Arguments**
+
+* `input` *(any)* input data structure.
+* `path` *(string)* input data structure path to the array to push the values to.
+* `values` *(function)* a sequence of values that is to be pushed to the input array.
```
Comment (ennovum) [inline README.md:93]: Not `pushed`, mr copypastor.

File: src/immutably-array/shift.test.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        const resultOutput = shift(input, null);
+        const expectedOutput = [];
+
+        testCase.deepEqual(resultOutput, expectedOutput);
+        testCase.notEqual(input, resultOutput);
+    };
+
+    testCase.doesNotThrow(testScenario1);
+
+    const testScenario2 = () => {
```
Comment (ennovum) [inline src/immutably-array/shift.test.js:17]: Isn't it exactly the same as `testScenario1`?

Comment (michalbe) [reply src/immutably-array/shift.test.js:17]: It indeed is Sir, exactly like `testScenarios1` & `testScenarios2` from `pop.test.js` this was inspired by.

**Pull request merged**